src/overlays/actors/ovl_Bg_Spot16_Doughnut/z_bg_spot16_doughnut.c

```diff
@@ -21,15 +30,136 @@ const ActorInit Bg_Spot16_Doughnut_InitVars = {
     (ActorFunc)BgSpot16Doughnut_Update,
     (ActorFunc)BgSpot16Doughnut_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Doughnut/BgSpot16Doughnut_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot16_Doughnut/BgSpot16Doughnut_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(unk_F4, 5500, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 5000, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 5000, ICHAIN_STOP),
+};
+
+static s16 sScales[] = {
+    0, 0, 70, 210, 300,
+};
+
+extern Gfx D_06000660[];
+extern Gfx D_06000FC0[];
+
+void BgSpot16Doughnut_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot16Doughnut* this = THIS;
+    s32 params;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Actor_SetScale(&this->actor, 0.10000000149011612f);
+    this->fireFlag = 0;
+    this->envColorAlpha = 0xFF;
+    params = this->actor.params;
+    if (params == 1 || params == 2 || params == 3 || params == 4) {
+        Actor_SetScale(&this->actor, sScales[this->actor.params] * 9.999999747378752e-05f);
```

Comment:
There is definitely a better way to write this float, but I'm not sure what would work without trying things directly.